src/StoreApi/Routes/V1/CheckoutOrder.php

```diff
@@ -0,0 +1,432 @@
+<?php
+namespace Automattic\WooCommerce\StoreApi\Routes\V1;
+
+use Automattic\WooCommerce\StoreApi\Payments\PaymentContext;
+use Automattic\WooCommerce\StoreApi\Payments\PaymentResult;
+use Automattic\WooCommerce\StoreApi\Exceptions\InvalidStockLevelsInCartException;
+use Automattic\WooCommerce\StoreApi\Exceptions\RouteException;
+use Automattic\WooCommerce\StoreApi\Utilities\OrderAuthorizationTrait;
+
+/**
+ * CheckoutOrder class.
+ */
+class CheckoutOrder extends AbstractCartRoute {
+	use OrderAuthorizationTrait;
+
+	/**
+	 * The route identifier.
+	 *
+	 * @var string
+	 */
+	const IDENTIFIER = 'checkout-order';
+
+	/**
+	 * The routes schema.
+	 *
+	 * @var string
+	 */
+	const SCHEMA_TYPE = 'checkout-order';
+
+	/**
+	 * Holds the current order being processed.
+	 *
+	 * @var \WC_Order
+	 */
+	private $order = null;
+
+	/**
+	 * Get the path of this REST route.
+	 *
+	 * @return string
+	 */
+	public function get_path() {
+		return '/checkout/(?P<id>[\d]+)';
+	}
+
+	/**
+	 * Checks if a nonce is required for the route.
+	 *
+	 * @param \WP_REST_Request $request Request.
+	 * @return bool
+	 */
+	protected function requires_nonce( \WP_REST_Request $request ) {
+		return true;
+	}
```

Comment:
Maybe there's no need for this method because https://github.com/woocommerce/woocommerce-blocks/blob/e0c7440ddc3fa897f3fc984845459aa472c4c42b/src/StoreApi/Routes/V1/AbstractCartRoute.php#L221-L223 already checks for the nonce if the request is not `GET`, this method would make no difference.

Reply:
Good catch!  Here's the fix [c8b12c7](https://github.com/woocommerce/woocommerce-blocks/pull/10287/commits/c8b12c7d86253c636ed8ee0b7249a57df1cf9c57).